packages/truffle-db/src/workspace/test/source.spec.ts

```diff
@@ -37,7 +37,8 @@ describe("Source", () => {
   });
 
   test("can be queried", async() => {
-    const getSourceResult = await wsClient.execute(GetSource, { id: expectedId });
+    const getSourceResult = await wsClient.executePersisted(GetSource, { id: expectedId });
+
```

Comment:
Maybe add another test to assert that `executePersisted` behavior in fact fails if you delete the persistence directory

Reply:
This is actually slightly more problematic than we thought -- I believe jsondownpouch actually saves the data in memory *and* in json format... so deleting the directory doesn't cause the test to fail. Of course, that means this way of testing persistence isn't accurate, so I'm a bit at a loss of how else to test it. I suppose I can check for the existence of the expected subdirectories?

Reply:
This is figured out with my last commit. Main issue appeared to be that if we instantiate two workspaces at the same time there appears to be some cross-talk between them? I'm not totally sure why but if I instantiate persistedWorkspace later on my tests work as expected.